File: turms-client-dart/lib/src/model/notification.dart
import 'package:fixnum/fixnum.dart';

import '../../turms_client.dart';

class Notification {
  final DateTime timestamp;
  final Int64 requesterId;
  final TurmsRequest relayedRequest;

  Notification(this.timestamp, this.requesterId, this.relayedRequest);

  @override
  int get hashCode =>
      timestamp.hashCode ^ requesterId.hashCode ^ relayedRequest.hashCode;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Notification &&
          timestamp == other.timestamp &&
          requesterId == other.requesterId &&
          relayedRequest == other.relayedRequest;

  @override
  String toString() =>
      'Notification{timestamp: $timestamp, requesterId: $requesterId, relayedRequest: $relayedRequest}';
}

File: turms-client-swift/Sources/TurmsClient/Model/Notification.swift
import Foundation

public struct Notification {
    public let timestamp: Date
    public let requesterId: Int64
    public let relayedRequest: TurmsRequest

    public init(timestamp: Date, requesterId: Int64, relayedRequest: TurmsRequest) {
        self.timestamp = timestamp
        self.requesterId = requesterId
        self.relayedRequest = relayedRequest
    }
}
